docs/agent-api.asciidoc

```diff
@@ -324,14 +324,82 @@ Whether or not the agent should monitor for uncaught exceptions and send them to
 
 Set this option to `false` to disable capture of stack traces for measured traces during instrumentation.
 
-[[source-context]]
-===== `sourceContext`
+[[source-context-error]]
+===== `sourceContextErrorAppFrames` + `sourceContextErrorLibraryFrames`
 
-* *Type:* Boolean
-* *Default:* `true`
-* *Env:* `ELASTIC_APM_SOURCE_CONTEXT`
+When an error is captured by the agent,
+its stack trace is stored in Elasticsearch.
+
+By default the agent will also collect a few lines of source code around the lines for each frame in the stack trace.
+This can make it easier to determine the cause of an error as the source code related to the error is visible directly in Kibana.
```

Comment:
I try to avoid referencing the X-Pack UI, as some users might not have installed it.

Maybe that's the wrong approach and we should remind users of the awesomeness that is the  X-Pack APM UI at every possibility. Worth discussing :)

Reply:
You can see it in Discover as well, so I didn't think that mentioning Kibana is mentioning X-Pack?

Reply:
In Discover you see a deeply nested JSON blob that, with a lot of imagination, can give you the source lines...

Reply:
No it's not ideal, but the data _is_ there and users can extract it in other fashions as well. I don't agree that this is equivalent to mentioning X-Pack (which I agree we shouldn't). But I'd love input on how to formulate this better if you have any suggestions - just couldn't find a better solution my self 😃 

Brainstorming: If the UX really is that terrible as you say for non-X-Pack users, should we consider disabling source code context completely unless you were an X-Pack user? The agent will of course not know this, so this would suggest that our default should be to set all these config options to `0` and then have a "Recommended setup for X-Pack users" document.

Reply:
It's hard for me to visualize how this looks but could you be slightly less specific and simply remove "as the source code related to the error is visible directly in Kibana." It sort of makes sense already that having some context is useful.

Another option is that you could show an example of the event (but then if the event structure is likely to change, you'll need to remember to update the example).

Reply:
@dedemorton removing the last part of the sentence is actually not a bad solution